micro-rdk/src/esp32/analog.rs

```diff
@@ -1,42 +1,33 @@
 #![allow(dead_code)]
 use crate::common::analog::{AnalogError, AnalogReader, AnalogResolution};
-use crate::esp32::esp_idf_svc::hal::adc::{AdcChannelDriver, AdcDriver};
+use crate::esp32::esp_idf_svc::hal::{
+    adc::oneshot::{AdcChannelDriver, AdcDriver},
+    gpio::ADCPin,
+};
+use std::borrow::Borrow;
 
-// TODO(RSDK-10188): Update to ESP-IDF 5 ADC API
-use crate::esp32::esp_idf_svc::hal::gpio::ADCPin;
-use std::sync::{Arc, Mutex};
-
-pub struct Esp32AnalogReader<'a, const A: u32, T: ADCPin> {
-    channel: AdcChannelDriver<'a, A, T>,
-    driver: Arc<Mutex<AdcDriver<'a, T::Adc>>>,
+pub struct Esp32AnalogReader<'a, T: ADCPin, M: Borrow<AdcDriver<'a, T::Adc>>> {
```

Comment:
I don't like the increase of complexity of types parameters, is there a way we can make it more readable. Like storing only the driver and building the channel on each read? Is building a channel expensive?

Reply:
I think that may actually make the type even more complex. 

At the very least the `new` input parameters would either need to be:
1. `(name: String, driver: AdcDriver<'d, T::Adc>, pin: impl Peripheral<P = T> + 'd,
    config: AdcChannelConfig)` if we want to create the `AdcChannelDriver` on-demand
2. `(name: String, adc: impl Adc, pin: impl Peripheral<P = T> + 'd, config: AdcChannelConfig)` if we want to handle both the `AdcDriver` and `AdcChannelDriver` initialization in the `Esp32AnalogReader`.